packages/gitgraph-js/src/gitgraph.ts

```diff
@@ -446,164 +544,128 @@ function createGitgraph(
-    if (gitgraph.isVertical && !commit.style.hasTooltipInCompactMode) return;
-
-    const tooltip = commit.renderTooltip
-      ? commit.renderTooltip(commit)
-      : createTooltip(commit);
-
-    $tooltip = createG({
-      translate: getWithCommitOffset(commit),
-      children: [tooltip],
-    });
+    function setChildrenForeignObjectHeight(node: Node): void {
+      if (node.nodeName === "foreignObject") {
+        // We have to access the first child's parentElement to retrieve
+        // the Element instead of the Node => we can compute dimensions.
+        const foreignObject = node.firstChild && node.firstChild.parentElement;
+        if (!foreignObject) return;
 
-    svg.firstChild.appendChild($tooltip);
-  }
+        // Force the height of the foreignObject (browser issue)
+        foreignObject.setAttribute(
+            "height",
+            getMessageHeight(foreignObject.firstElementChild).toString(),
+        );
+      }
 
-  function getWithCommitOffset({ x, y }: Coordinate): Coordinate {
-    return { x, y: commitYWithOffsets[y] || y };
+      node.childNodes.forEach(setChildrenForeignObjectHeight);
+    }
   }
 
-  function setBranchLabelRef(commit: Commit, branchLabels: SVGGElement): void {
-    if (!commitsElements[commit.hashAbbrev]) {
-      initCommitElements(commit);
+  protected setBranchLabelRef(commit: Commit, branchLabels: SVGGElement): void {
+    if (!this.commitsElements[commit.hashAbbrev]) {
+      this.initCommitElements(commit);
     }
 
-    commitsElements[commit.hashAbbrev].branchLabel = branchLabels;
+    this.commitsElements[commit.hashAbbrev].branchLabel = branchLabels;
   }
 
-  function setMessageRef(commit: Commit, message: SVGGElement | null): void {
-    if (!commitsElements[commit.hashAbbrev]) {
-      initCommitElements(commit);
+  protected setMessageRef(commit: Commit, message: SVGGElement | null): void {
+    if (!this.commitsElements[commit.hashAbbrev]) {
+      this.initCommitElements(commit);
     }
 
-    commitsElements[commit.hashAbbrev].message = message;
+    this.commitsElements[commit.hashAbbrev].message = message;
   }
 
-  function setTagRef(commit: Commit, tag: SVGGElement): void {
-    if (!commitsElements[commit.hashAbbrev]) {
-      initCommitElements(commit);
+  protected setTagRef(commit: Commit, tag: SVGGElement): void {
+    if (!this.commitsElements[commit.hashAbbrev]) {
+      this.initCommitElements(commit);
     }
 
-    commitsElements[commit.hashAbbrev].tags.push(tag);
+    this.commitsElements[commit.hashAbbrev].tags.push(tag);
   }
 
-  function initCommitElements(commit: Commit): void {
-    commitsElements[commit.hashAbbrev] = {
+  protected initCommitElements(commit: Commit): void {
+    this.commitsElements[commit.hashAbbrev] = {
       branchLabel: null,
       tags: [],
       message: null,
     };
   }
 }
 
+function createGitgraph(
+  graphContainer: HTMLElement,
+  options?: GitgraphOptions,
+  renderer?: Renderer,
```

Comment:
I can see we're adding the possibility to provide a custom renderer, but there's no test associated that would present the new capabilities. 

Could you add one to illustrate what is possible to do?

I think that will help explain the benefits of such a change. Also, it will make sure we don't break these new features next time we touch the code (e.g. we refactor the `Renderer` class and break the public API without realizing it).

Reply:
will do.